### PR TITLE
fix(rest): New endpoint to download license archive files in admin page

### PR DESCRIPTION
**Pull request by nikkuma7**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
> * Did you add or update any new dependencies that are required for your change?

Issue: #2105 

### How To Test?
URL:  /resource/api/licenses/downloadLicenses
1: user should have admin access.
2: Go to admin tab.
3: download licenses archive.

> How should these changes be tested by the reviewer?
> Have you implemented any additional tests?

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by rudra-superrr** [comment]
Rest docs broken for your endpoint.
image

File: rest/resource-server/src/docs/asciidoc/licenses.adoc [hunk 1]
```diff
@@ -108,4 +108,22 @@ A `POST` request will import SPDX info.
 include::{snippets}/should_document_import_spdx_info/curl-request.adoc[]
 
 ===== Example response
-include::{snippets}/should_document_import_spdx_info/http-response.adoc[]
\ No newline at end of file
+include::{snippets}/should_document_import_spdx_info/http-response.adoc[]
+
+
+[[resources-admin-download--licenses-archive-file_req]]
+==== Downloading licenses archive folder for admin request
```
Comment (rudra-superrr) [inline rest/resource-server/src/docs/asciidoc/licenses.adoc:115]: Change it to Download License Archive on line 114 & 115.

File: rest/resource-server/src/docs/asciidoc/licenses.adoc [hunk 2]
```diff
@@ -108,4 +108,22 @@ A `POST` request will import SPDX info.
...
+
+A `GET` request help to download the licenses archive folder for admin user.
+
```
Comment (rudra-superrr) [inline rest/resource-server/src/docs/asciidoc/licenses.adoc:118]: Change it to, A `GET` request to download license archive

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/LicenseController.java [hunk 3]
```diff
@@ -37,11 +38,19 @@
...
+import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Set;
+import java.util.zip.ZipOutputStream;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/LicenseController.java:54]: remove unused imports.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/LicenseController.java [hunk 4]
```diff
@@ -53,6 +62,9 @@ public class LicenseController implements RepresentationModelProcessor<Repositor
 
     @NonNull
     private final Sw360LicenseService licenseService;
+    
+    @NonNull
+    private final Sw360LicenseInfoService licenseInfoService;
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/LicenseController.java:68]: remove unused object.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java [hunk 5]
```diff
@@ -11,31 +11,50 @@
...
+import java.nio.ByteBuffer;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
+import java.util.zip.ZipOutputStream;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java:58]: remove unused imports.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java [hunk 6]
```diff
@@ -44,6 +63,8 @@
 public class Sw360LicenseService {
     @Value("${sw360.thrift-server-url:http://localhost:8080}")
     private String thriftServerUrl;
+    private Sw360LicenseInfoService  licenseInfoService;
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java:66]: remove unused object.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java [hunk 7]
```diff
@@ -161,4 +182,39 @@ public void importSpdxInformation(User sw360User) throws TException {
             throw new HttpMessageNotReadableException("Unable to import All Spdx license. User is not admin");
         }
     }
+    
+	public void getDownloadLicenses(User sw360User ,HttpServletRequest request,HttpServletResponse response) throws TException,IOException{
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java:186]: Use proper indentation for this method.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java [hunk 8]
```diff
@@ -161,4 +182,39 @@ public void importSpdxInformation(User sw360User) throws TException {
...
+		if (!PermissionUtils.isUserAtLeast(UserGroup.ADMIN, sw360User)) {
+			throw new HttpMessageNotReadableException("Unable to download archive license. User is not admin");
+		}
+		try {
+		    LicenseService.Iface sw360LicenseClient = getThriftLicenseClient();
+		    String fileConstant="LicensesBackup.lics";
+		    Map<String, InputStream> fileNameToStreams = (new LicsExporter(sw360LicenseClient)).getFilenameToCSVStreams();
+		    final ByteArrayOutputStream outB = new ByteArrayOutputStream();
```
Comment (rudra-superrr) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java:194]: use proper variable name for outB

File: rest/resource-server/src/docs/asciidoc/licenses.adoc (hunk 2 above)
Comment (nikkuma7) [reply rest/resource-server/src/docs/asciidoc/licenses.adoc:116]: Done.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java (hunk 5 above)
Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java:58]: we do not have unused import.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java (hunk 7 above)
Comment (nikkuma7) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/license/Sw360LicenseService.java]: Done.

**Comment by rudra-superrr** [comment]
code looks good.

**Comment by rudra-superrr** [comment]
Testing was successful.

**Pull request merged**